cloudinit/net/__init__.py

```diff
@@ -971,18 +972,38 @@ def get_ib_hwaddrs_by_interface():
     return ret
 
 
-def has_url_connectivity(url):
-    """Return true when the instance has access to the provided URL
+def has_url_connectivity(
+    url_data: Union[MutableMapping[str, Any], str]
```

Comment:
@TheRealFalcon why do you think we need to support either str or MutableMapping? We only hae [one call-site to has_url_connectivity in cloudinit.net](https://github.com/canonical/cloud-init/blob/main/cloudinit/net/__init__.py#L1070) I think. Any reason not to just mandate a Mapping param and avoid trying to be too flexible with different param types?

Reply:
Also for background,  I think Vultr is the only datasource currently setting "connectivity_url" for the EphemeralDHCP contextmanager.